idem_azurerm/exec/azurerm/containerregistry/registry.py

```diff
@@ -0,0 +1,891 @@
+                "resource_id": source_resource_id,
+                "registry_uri": source_registry_uri,
+                "credentials": source_credentials,
+            },
+            target_tags=target_tags,
+            untagged_target_repositories=untagged_target_repositories,
+            mode=mode,
+            **kwargs,
+        )
+    except TypeError as exc:
+        result = {
+            "error": "The object model could not be built. ({0})".format(str(exc))
+        }
+        return result
+
+    try:
+        ret = regconn.registries.import_image(
+            registry_name=name,
+            resource_group_name=resource_group,
+            parameters=importmodel,
+        )
+        ret.wait()
+        result = True
+    except (CloudError, SerializationError) as exc:
+        await hub.exec.azurerm.utils.log_cloud_error(
+            "containerregistry", str(exc), **kwargs
+        )
+        result = {"error": str(exc)}
+
+    return result
+
+
+async def schedule_run(
+    hub,
+    ctx,
+    name,
+    resource_group,
+    run_type,
+    is_archive_enabled=None,
+    task_name=None,
+    task_file_path=None,
+    values_file_path=None,
+    encoded_task_content=None,
+    encoded_values_content=None,
+    image_names=None,
+    is_push_enabled=None,
+    no_cache=None,
+    target=None,
+    values_dict=None,
+    timeout=None,
+    platform_os=None,
+    platform_arch=None,
+    platform_variant=None,
+    agent_num_cores=None,
+    source_location=None,
+    credential_login_mode=None,
+    credential_login_server=None,
+    username=None,
+    password=None,
+    **kwargs,
+):
+    """
+    .. versionadded:: 3.0.0
+
+    Schedules a new run based on the request parameters and add it to the run queue.
+
+    :param name: The name of the container registry.
+
+    :param resource_group: The name of the resource group to which the container registry belongs.
+
+    :param run_type: The type of run to be scheduled. Must be FileTaskRun, TaskRun, EncodedTaskRun, or DockerBuild.
+
+    :param is_archive_enabled: The value that indicates whether archiving is enabled for the run or not.
+
+    :param task_name: (TaskRun REQUIRED) The name of task against which run has to be queued.
+
+    :param task_file_path: (FileTaskRun, DockerBuild REQUIRED) The template/definition file (or Dockerfile) path
+        relative to the source.
+
+    :param values_file_path: (FileTaskRun) The values/parameters file path relative to the source.
+
+    :param encoded_task_content: (EncodedTaskRun REQUIRED) Base64 encoded value of the template/definition file content.
+
+    :param encoded_values_content: (DockerBuild) Base64 encoded value of the parameters/values file content.
+
+    :param image_names: (DockerBuild) A list of strings containing the fully qualified image names including the
+        repository and tag.
+
+    :param is_push_enabled: (DockerBuild) The value of this property indicates whether the image built should be pushed
```

Comment:
For uniformity sake I would add REQUIRED to every requirement in parenthesis instead of having some with it and some without it.

Reply:
Not all of them ARE required...